test/e2e/testdata/acme/certificate_gcp.yaml

```diff
@@ -0,0 +1,14 @@
+apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  name: cert-with-acme-dns01-clouddns-ambient
+  namespace: NAMESPACE
+spec:
+  secretName: cert-with-acme-dns01-clouddns-ambient
+  issuerRef:
+    # The issuer created previously
+    kind: ClusterIssuer
+    name: acme-dns01-clouddns-ambient
+  dnsNames:
+    - RANDOM_STR.qe1.gcp.devcluster.openshift.com
+    - '*.RANDOM_STR.qe1.gcp.devcluster.openshift.com'
```

Comment:
Can you also make `qe1.gcp.devcluster.openshift.com` a placeholder like `DNS_NAME` instead.
And at runtime you could replace the same using your `replaceStrInFile` func with value of `library.GetClusterBaseDomain(ctx, configClient)`.

Something like this, inside your code of the test case:
```go
baseDomain, err := library.GetClusterBaseDomain(ctx, configClient)
Expect(err).NotTo(HaveOccurred())

oldStr := []string{"RANDOM_STR", "DNS_NAME"}
newStr := []string{randomString, baseDomain}
// ...
```
This would ensure that when these tests are run by someone else on a non-QE infra, it still works out for them without changes!

Reply:
updated.

Reply:
@swghosh thanks! Actually I am aware of the need of replacing DNS_NAME. But test/e2e/testdata/acme/clusterissuer_gcp.yaml has `project: openshift-qe`.
```
        cloudDNS:
          # The ID of the GCP project
          project: openshift-qe
```
The ID of the GCP project needs to be replaced too, but seems not easy to query it in programming, so I directly just did not raise DNS_NAME replacement. @swghosh @geliu2016 @lunarwhite , any idea? Or just let as is for now?

Reply:
Good catch @xingxingxia, I missed it in earlier reviews.

You can use similar replacement for project ID, too.
```
infra, err := configClient.Infrastructures().Get(ctx, "cluster", metav1.GetOptions{})
Expect(err).NotTo(HaveOccurred())

gcpProjectId := infra.Status.PlatformStatus.GCP.ProjectID
Expect(gcpProjectId).NotTo(Equal(""))

// replace "GCP_PROJECT_ID" in file with the value ...
```

Reply:
As `configClient` is already used in another test case: 
```
ACME Certificate
dns-01 challenge using ambient credentials
should obtain a valid LetsEncrypt certificate
```

> The ID of the GCP project needs to be replaced too, but seems not easy to query it in programming

so guessing it should be easy to do :)

Reply:
updated.